Odata-docs/webapi-8/tutorials/basic-crud-in-fsharp.md

```diff
@@ -0,0 +1,431 @@
+:white_check_mark: Add an OData controller  
+:white_check_mark: Run the OData service  
+:white_check_mark: Request entity collection  
+:white_check_mark: Request a single entity  
+:white_check_mark: Create an entity  
+:white_check_mark: Update an entity  
+:white_check_mark: Delete an entity  
+
+## Prerequisites
+
+[!INCLUDE[](../../includes/appliesto-webapi-v8-net-prereqs-vs.md)]
+
+## Create an ASP.NET Core application
+
+- Start Visual Studio 2022 and select **Create a new project**.
+- In the **Create a new project** dialog:
+  - Enter `Empty` in the **Search for templates** search box.
+  - Select **ASP.NET Core Empty** project template and select **Next**.
+
+    ![Screenshot of creating aspnetcore empty F# project using vs 2022 targeting net 6.0 framework](../../assets/2023-08-15-aspnet-core-odata-8-create-empty-fsharp-project-vs2022.png)
+- Name the project _BasicCrudInFSharp_ and select **Next**.
+- In the **Additional information** dialog: 
+  - Select **.NET 6.0 (Long Term Support)**.
+  - Uncheck **Configure for HTTPS** checkbox.
+  - Select **Create**.
+
+  ![Screenshot of creating aspnetcore F# project using vs 2022 targeting net core 6.0 framework additional info](../../assets/2023-08-15-aspnet-core-odata-8-create-fsharp-project-vs2022-additional-info.png)
+
+## Install required packages
+
+Install the following Nuget package:
+- [Microsoft.AspNetCore.OData](https://www.nuget.org/packages/Microsoft.AspNetCore.OData) 8.x
+
+# [Visual Studio](#tab/visual-studio)
+
+In the Visual Studio **Package Manager Console**:
+
+```powershell
+Install-Package Microsoft.AspNetCore.OData
+```
+
+# [.NET Core CLI](#tab/netcore-cli)
+
+```dotnetcli
+dotnet add package Microsoft.AspNetCore.OData
+```
+
+---
+
+## Add data models
+Add a folder named **Models** to the project and then add the following type:
+
+**`Person` class**
+
+```fsharp
+namespace BasicCrudInFSharp.Models
+
+type Person() = 
+    member val Id = 0 with get, set
+    member val Name = "" with get, set
+    member val Age = 0 with get, set
+```
+
+## Add an in-memory collection as data source
+Add an in-memory collection to serve as the data source for the OData service:
+
+```fsharp
+namespace BasicCrudInFSharp.Data
+
+open BasicCrudInFSharp.Models
+
+module DataSource =
+    let private GetPeople() =
+        let people = ResizeArray<Person>()
+
+        people.Add(new Person(Id = 1, Name = "Sue", Age = 19))
+        people.Add(new Person(Id = 2, Name = "Joe", Age = 17))
+        people.Add(new Person(Id = 3, Name = "Luc", Age = 23))
+
+        people
+
+    let People = GetPeople()
+```
+
+**NOTE:** The F# compiler requires files (and code within a file) to be organized in dependency order. In our project for example, the compiler has to "see" the `Person` type first before the `DataSource` module since the latter is dependent on the former. For that reason, you'll need to manually edit the _.fsproj_ file to reorder the files as follows:
```

Comment:
In this tutorial you create the `Person` type first then the `DataSource` module. Yet you still have to manually re-order the project file such that Person appears before DataSource. Does that mean that Visual Studio does not add compilation items in the order you create them?

Reply:
@habbes "Does that mean that Visual Studio does not add compilation items in the order you create them?" - It'd seem so. Visual Studio could be ordering them alphabetically